### PR TITLE
Add countitem(item_id) to Perl/Lua.

### PR DESCRIPTION
**Pull request by KinglyKrab**

### CODE REVIEW

File: zone/questmgr.cpp [hunk 1]
```diff
@@ -2580,6 +2580,32 @@ int QuestManager::collectitems(uint32 item_id, bool remove)
...
+		{ EQEmu::invslot::POSSESSIONS_BEGIN, EQEmu::invslot::POSSESSIONS_END },
+		{ EQEmu::invbag::GENERAL_BAGS_BEGIN, EQEmu::invbag::GENERAL_BAGS_END },
+		{ EQEmu::invbag::CURSOR_BAG_BEGIN, EQEmu::invbag::CURSOR_BAG_END},
+		{ EQEmu::invslot::BANK_BEGIN, EQEmu::invslot::BANK_END },
+		{ EQEmu::invbag::BANK_BAGS_BEGIN, EQEmu::invbag::BANK_BAGS_END },
+		{ EQEmu::invslot::SHARED_BANK_BEGIN, EQEmu::invslot::SHARED_BANK_END },
+		{ EQEmu::invbag::SHARED_BANK_BAGS_BEGIN, EQEmu::invbag::SHARED_BANK_BAGS_END },
+	};
+	
+	for (int slot_index = 0; slot_index < 7; ++slot_index) {
```
Comment (mackal) [inline zone/questmgr.cpp]: No need to hardcode the 7 here, can use sizeof to get the size.

File: zone/questmgr.cpp [hunk 2]
```diff
@@ -2580,6 +2580,32 @@ int QuestManager::collectitems(uint32 item_id, bool remove)
...
 
+int QuestManager::countitem(uint32 item_id) {
+	QuestManagerCurrentQuestVars();
+	int quantity = 0;
+	EQEmu::ItemInstance *item = nullptr;
+	static const int16 slots[][2] = {
+		{ EQEmu::invslot::POSSESSIONS_BEGIN, EQEmu::invslot::POSSESSIONS_END },
+		{ EQEmu::invbag::GENERAL_BAGS_BEGIN, EQEmu::invbag::GENERAL_BAGS_END },
+		{ EQEmu::invbag::CURSOR_BAG_BEGIN, EQEmu::invbag::CURSOR_BAG_END},
+		{ EQEmu::invslot::BANK_BEGIN, EQEmu::invslot::BANK_END },
```
Comment (alimalkhalifa) [inline zone/questmgr.cpp]: Do we always want to check bank here?

I would imagine most use cases care only about what is in the current inventory

File: zone/questmgr.h [hunk 3]
```diff
@@ -219,6 +219,7 @@ class QuestManager {
     int getlevel(uint8 type);
     int collectitems(uint32 item_id, bool remove);
     int collectitems_processSlot(int16 slot_id, uint32 item_id, bool remove);
+	int countitem(uint32 item_id);
```
[the remark this answers is not in the document]
Comment (KinglyKrab) [reply zone/questmgr.h]: The whole file is weirdly formatted like that, not sure.

File: zone/questmgr.cpp (hunk 2 above)
Comment (KinglyKrab) [reply zone/questmgr.cpp]: You can use quest::collectitems(itemid, remove?) if you want just in inventory.

This method is specifically to count any/all occurrences of an item in a player's inventory.

Comment (alimalkhalifa) [reply zone/questmgr.cpp]: Makes sense!

File: zone/questmgr.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (alimalkhalifa) [reply zone/questmgr.h]: Yikes 😑

**Pull request merged**